forge-game/src/main/java/forge/game/ability/effects/CloneEffect.java

```diff
@@ -202,4 +203,27 @@ public void run() {
         }
     }
 
+    protected void addPumpUntil(SpellAbility sa, final Card c, long timestamp) {
```

Comment:
Shouldn't need to duplicate that code
You could just make the one in TokenEffectBase be static or something